### PR TITLE
Unit tests for checking to make sure we can filter HK data on app source 

### PR DESCRIPTION
**Pull request by rickpasetto**
Moved the unit tests to LoopKit where they can use **LoopKit Example** (which has HK entitlements) as a host app.  

I also had to create a new `LoopKitHostedTests` target for running these HK tests.  The reason this was needed was because the HKFiltering tests need to run in a "host" app that has the HealthKit entitlement.  **LoopKit Example** has that.  However, we couldn't just make all of `LoopKitTests` use that hosted app because it causes other problems.

(Thanks again to Darin for pointing this out)

Note this now makes https://github.com/tidepool-org/Loop/pull/126 invalid, closed it.

### CODE REVIEW

File: LoopKitTests/Extensions/HKHealthStoreMock.swift [hunk 1]
```diff
@@ -45,6 +46,12 @@ class HKHealthStoreMock: HKHealthStore {
     }
 }
 
+extension HKHealthStoreMock {
+
+    override func execute(_ query: HKQuery) {
+        self.lastQuery = query
```
Comment (darinkrauss) [inline LoopKitTests/Extensions/HKHealthStoreMock.swift:52]: Do you need to call `super.execute(query)`?

File: LoopKitHostedTests.plist [hunk 2]
```diff
@@ -0,0 +1,24 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (darinkrauss) [inline LoopKitHostedTests.plist]: Needs to be pushed into a `LoopKitHostedTests` directory along with all of the tests.

Comment (darinkrauss) [reply LoopKitHostedTests.plist]: Should mirror the `LoopKitTests` directory and structure, I think.

File: LoopKitTests/CarbStoreHKFilterTests.swift [hunk 3]
```diff
@@ -0,0 +1,35 @@
+//
+//  CarbStoreHKFilterTests.swift
+//  LoopTests
```
Comment (darinkrauss) [inline LoopKitTests/CarbStoreHKFilterTests.swift]: These tests belong in a `LoopKitHostedTests` directory and this name should be updated to reflect that.

File: LoopKit/DosingDecisionStore.swift [hunk 4]
```diff
@@ -9,7 +9,6 @@
 import os.log
 import Foundation
 import CoreData
-import HealthKit
```
Comment (darinkrauss) [inline LoopKit/DosingDecisionStore.swift:12]: [nit] Not relevant to this PR, but okay.

File: LoopKitTests/CarbStoreHKFilterTests.swift [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+
+    func testEntriesFromAllSources() {
+        checkEntries(fromCurrentAppOnly: false)
+    }
+    
+    func testEntriesFromCurrentAppOnly() {
+        checkEntries(fromCurrentAppOnly: true)
+    }
+
+    private func checkEntries(fromCurrentAppOnly: Bool, file: StaticString = #file, line: UInt = #line) {
```
Comment (darinkrauss) [inline LoopKitTests/CarbStoreHKFilterTests.swift]: [nit] I would drop the `file` and `line` parameters. I think you want the actual `XCT*` to indicate their actual current file/line. When run, the test name itself will indicate which test failed.

File: LoopKit.xcodeproj/project.pbxproj [hunk 6]
```diff
@@ -2587,6 +2742,108 @@
 /* End PBXProject section */
 
 /* Begin PBXResourcesBuildPhase section */
+		1DEE229E24A676A300693C32 /* Resources */ = {
+			isa = PBXResourcesBuildPhase;
+			buildActionMask = 2147483647;
+			files = (
+				1DEE229F24A676A300693C32 /* carbs_on_board_output.json in Resources */,
```
Comment (darinkrauss) [inline LoopKit.xcodeproj/project.pbxproj]: [nit] What is all of this? It doesn't seem relevant to this PR.

File: LoopKit.xcodeproj/project.pbxproj [hunk 7]
```diff
@@ -1929,13 +2061,13 @@
...
 				A95A1D882460F8930079378D /* PumpManagerStatusTests.swift */,
-				A99C7372233990D400C80963 /* TempBasalRecommendationTests.swift */,
-				8974AFBF22120D7A0043F01B /* TemporaryScheduleOverrideTests.swift */,
-				891A3FD02249948A00378B27 /* TemporaryScheduleOverrideHistoryTests.swift */,
 				434C5F9F209ABD4700B2FD1A /* QuantityFormatterTests.swift */,
 				43D8FE1C1C72906E0073BE78 /* QuantityScheduleTests.swift */,
-				A971C8A123C6B17D0099BEFC /* SettingsStoreTests.swift */,
 				A99C73782339ACDC00C80963 /* ServiceTests.swift */,
+				A971C8A123C6B17D0099BEFC /* SettingsStoreTests.swift */,
+				A99C7372233990D400C80963 /* TempBasalRecommendationTests.swift */,
```
Comment (darinkrauss) [inline LoopKit.xcodeproj/project.pbxproj:1981]: [nit] What is all of this? It doesn't seem relevant to this PR.

File: LoopKit.xcodeproj/project.pbxproj [hunk 8]
```diff
@@ -16,6 +16,106 @@
...
 		1DD3DEB624451C3300BD8E40 /* Alert.swift in Sources */ = {isa = PBXBuildFile; fileRef = 1DA649AA2445174400F61E75 /* Alert.swift */; };
+		1DEE226F24A676A300693C32 /* GlucoseStoreHKFilterTests.swift in Sources */ = {isa = PBXBuildFile; fileRef = 1DEE226024A6642300693C32 /* GlucoseStoreHKFilterTests.swift */; };
+		1DEE227724A676A300693C32 /* HKHealthStoreMock.swift in Sources */ = {isa = PBXBuildFile; fileRef = 437AFF1E203A763F008C4892 /* HKHealthStoreMock.swift */; };
+		1DEE228424A676A300693C32 /* CarbStoreHKFilterTests.swift in Sources */ = {isa = PBXBuildFile; fileRef = 1DEE226224A6642400693C32 /* CarbStoreHKFilterTests.swift */; };
+		1DEE229724A676A300693C32 /* DoseStoreHKFilterTests.swift in Sources */ = {isa = PBXBuildFile; fileRef = 1DEE226124A6642300693C32 /* DoseStoreHKFilterTests.swift */; };
+		1DEE229D24A676A300693C32 /* LoopKit.framework in Frameworks */ = {isa = PBXBuildFile; fileRef = 43D8FDCB1C728FDF0073BE78 /* LoopKit.framework */; };
+		1DEE229F24A676A300693C32 /* carbs_on_board_output.json in Resources */ = {isa = PBXBuildFile; fileRef = 43D8FE461C7291A60073BE78 /* carbs_on_board_output.json */; };
```
Comment (darinkrauss) [inline LoopKit.xcodeproj/project.pbxproj]: [nit] What is all of this? It doesn't seem relevant to this PR.

**Review by darinkrauss — COMMENTED** [review]
A number of comments for you to consider. If you still think it is fine, just bounce it back to me and I'll approve. Thanks!

**Review by darinkrauss — COMMENTED** [review]
A number of comments for you to consider. If you still think it is fine, just bounce it back to me and I'll approve. Thanks!

File: LoopKitTests/Extensions/HKHealthStoreMock.swift (hunk 1 above)
Comment (rickpasetto) [reply LoopKitTests/Extensions/HKHealthStoreMock.swift:52]: No...Intentionally.  The seam is intercepting the call to save the query only.

File: LoopKitHostedTests.plist (hunk 2 above)
Comment (rickpasetto) [reply LoopKitHostedTests.plist]: Huh...Xcode created this file, not sure why it was put in this directory location.  I'll fix it.

File: LoopKitTests/CarbStoreHKFilterTests.swift (hunk 5 above)
Comment (rickpasetto) [reply LoopKitTests/CarbStoreHKFilterTests.swift]: This is important to see where the test failure was in Xcode (i.e. the red line)

File: LoopKit.xcodeproj/project.pbxproj (hunk 6 above)
Comment (rickpasetto) [reply LoopKit.xcodeproj/project.pbxproj]: This is the result of creating a new target

File: LoopKit.xcodeproj/project.pbxproj (hunk 7 above)
Comment (rickpasetto) [reply LoopKit.xcodeproj/project.pbxproj:1981]: This is the result of creating a new target

File: LoopKit.xcodeproj/project.pbxproj (hunk 8 above)
Comment (rickpasetto) [reply LoopKit.xcodeproj/project.pbxproj]: This is the result of creating a new target

File: LoopKit/DosingDecisionStore.swift (hunk 4 above)
Comment (rickpasetto) [reply LoopKit/DosingDecisionStore.swift:12]: ah, I can remove it if you feel strongly...it's cruft from trying to debug the problem I was having primarily.

File: LoopKit.xcodeproj/project.pbxproj (hunk 8 above)
Comment (darinkrauss) [reply LoopKit.xcodeproj/project.pbxproj]: Since these files shouldn't be required for the new target, I think you can go to these files and remove them from the new target.

File: LoopKit.xcodeproj/project.pbxproj (hunk 6 above)
Comment (rickpasetto) [reply LoopKit.xcodeproj/project.pbxproj]: ah but somehow it copied all those bundled resources which are not needed.  I'll get rid of them.

File: LoopKit.xcodeproj/project.pbxproj [hunk 9]
```diff
@@ -3599,6 +3695,30 @@
 /* End PBXVariantGroup section */
 
 /* Begin XCBuildConfiguration section */
+		1DEE22FF24A676A300693C32 /* Debug */ = {
+			isa = XCBuildConfiguration;
+			buildSettings = {
+				DEVELOPMENT_TEAM = "";
+				INFOPLIST_FILE = LoopKitHostedTests/LoopKitHostedTests.plist;
+				LD_RUNPATH_SEARCH_PATHS = "$(inherited) @executable_path/Frameworks @loader_path/Frameworks";
+				PRODUCT_BUNDLE_IDENTIFIER = com.loopkit.LoopKitTests;
```
Comment (darinkrauss) [inline LoopKit.xcodeproj/project.pbxproj]: [nit] Might want to update the bundle identifier. Probably won't make a difference, but...

Comment (darinkrauss) [reply LoopKit.xcodeproj/project.pbxproj]: Same down below.

File: LoopKit/DosingDecisionStore.swift (hunk 4 above)
Comment (darinkrauss) [reply LoopKit/DosingDecisionStore.swift:12]: No worries.

File: LoopKitHostedTests/GlucoseStoreHKFilterTests.swift [hunk 10]
```diff
@@ -0,0 +1,35 @@
+//
+//  GlucoseStoreHKFilterTests.swift
+//  LoopTests
```
Comment (darinkrauss) [inline LoopKitHostedTests/GlucoseStoreHKFilterTests.swift]: [nit] LoopKitHostedTests?

Comment (darinkrauss) [reply LoopKitHostedTests/GlucoseStoreHKFilterTests.swift]: Same for the other files.

**Review by darinkrauss — APPROVED** [review]
Just a few minor nits you might want to cleanup, but otherwise LGTM! 🦔

File: LoopKitHostedTests/GlucoseStoreHKFilterTests.swift (hunk 10 above)
Comment (rickpasetto) [reply LoopKitHostedTests/GlucoseStoreHKFilterTests.swift]: wow you're eagle-eyed today 👍
